### PR TITLE
Preliminary work using ScheduleExecutorService

### PR DESCRIPTION
**Pull request by CTrando**

### CODE REVIEW

File: src/main/java/com/internhub/data/positions/extractors/PositionExtractor.java [hunk 1]
```diff
@@ -101,6 +101,10 @@ private Position tryGetPosition(Page page, Company company) {
     }
 
     private boolean isValidPage(Page page) {
+        if(page == null) {
```
Comment (CTrando) [inline src/main/java/com/internhub/data/positions/extractors/PositionExtractor.java]: Page can be null from getPage

File: src/main/java/com/internhub/data/positions/scrapers/strategies/impl/PositionBFSMTStrategy.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+import java.util.PriorityQueue;
+import java.util.Set;
+import java.util.concurrent.*;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+
+/**
+ * Had to do something like this to escape the schedule executor service task
+ */
+class ScheduledDelayFinishedException extends RuntimeException {
```
Comment (CTrando) [inline src/main/java/com/internhub/data/positions/scrapers/strategies/impl/PositionBFSMTStrategy.java]: If you have a better way to do this please tell me. I tried using a CountDownLatch initialized to max links, but then we waste time on companies that don't have that many links in their tree.

File: src/main/java/com/internhub/data/positions/scrapers/strategies/impl/PositionBFSMTStrategy.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+                () -> {
+                    if (!candidates.isEmpty() && totalLinks.get() < MAX_TOTAL_LINKS) {
+                        Candidate candidate = candidates.poll();
+                        logger.info(String.format(
+                                "[%d/%d] Visiting %s (depth = %d) ...",
+                                totalLinks.getAndIncrement(), MAX_TOTAL_LINKS, candidate.link, candidate.depth));
+
+                        processCandidate(company, candidate, candidates, visited, results);
+                    } else {
+                        throw new ScheduledDelayFinishedException();
```
Comment (CTrando) [inline src/main/java/com/internhub/data/positions/scrapers/strategies/impl/PositionBFSMTStrategy.java]: This is to break out of the scheduled task. Exceptions are slow so this isn't ideal.

**Comment by dmhacker** [comment]
Fixes issue #7.

**Pull request merged**